### ISSUE TITLE
Biosample query throws error to "contact Biopython developers"

### ISSUE THREAD

**Issue by camhilker** [biopython/biopython#2817]
### Setup

I am reporting a problem with Biopython version 1.74, Python version 3.7.6, and operating
system MacOS Catalina 10.15.3 as follows:

```python
from Bio import Entrez
import pprint

handle = Entrez.efetch(db="biosample", id="SAMEA3502652", rettype="xml", retmode="text")
record = Entrez.read(handle, validate=False)

pp = pprint.PrettyPrinter(indent=5)
pp.pprint(record)
```

### Error
```
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
~/Users/camelliahilker/miniconda3/envs/pubmed_project/lib/python3.7/site-packages/Bio/Entrez/Parser.py in read(self, handle)
    344         try:
--> 345             return self.record
    346         except AttributeError:

AttributeError: 'DataHandler' object has no attribute 'record'

During handling of the above exception, another exception occurred:

RuntimeError                              Traceback (most recent call last)
<ipython-input-69-82ee22455e92> in <module>
      2 
      3 handle = Entrez.efetch(db="biosample", id="SAMEA3502652", rettype="xml", retmode="text")
----> 4 record = Entrez.read(handle, validate=False)
      5 
      6 pp = pprint.PrettyPrinter(indent=5)

~/Users/camelliahilker/miniconda3/envs/pubmed_project/lib/python3.7/site-packages/Bio/Entrez/__init__.py in read(handle, validate, escape)
    472     from .Parser import DataHandler
    473     handler = DataHandler(validate, escape)
--> 474     record = handler.read(handle)
    475     return record
    476 

~/Users/camelliahilker/miniconda3/envs/pubmed_project/lib/python3.7/site-packages/Bio/Entrez/Parser.py in read(self, handle)
    349                 # any errors, so self.record should be defined. If not, this is
    350                 # a bug.
--> 351                 raise RuntimeError("Failed to parse the XML file correctly, possibly due to a bug in Bio.Entrez. Please contact the Biopython developers via the mailing list or GitHub for assistance.")
    352             else:
    353                 # We did not see the initial <!xml declaration, so probably

RuntimeError: Failed to parse the XML file correctly, possibly due to a bug in Bio.Entrez. Please contact the Biopython developers via the mailing list or GitHub for assistance.
```

### Expected behaviour

Hello, I am trying to extract information about a list of samples by using the biosample IDs via Entrez API (biosample database).

### Actual behaviour

Throws error. How can I successfully query this database?

### Steps to reproduce

**Comment by mdehoon** [comment]
Short explanation:
Bio.Entrez cannot parse data from the biosample database because the XML data returned by NCBI Entrez does not include a DTD.
Your options are either to write to NCBI Entrez and ask them to fix this, or alternatively to use one of the generic XML parsers in Python (probably `ElementTree`) to parse the data returned by NCBI Entrez.

Long explanation:

Typically, XML data returned by NCBI Entrez look like this:
```
<?xml version="1.0" ?>
<!DOCTYPE PubmedArticleSet PUBLIC "-//NLM//DTD PubMedArticle, 1st January 2019//EN" "https://dtd.nlm.nih.gov/ncbi/pubmed/out/pubmed_190101.dtd">
<PubmedArticleSet>
<PubmedArticle>
    <MedlineCitation Status="MEDLINE" Owner="NLM">
        <PMID Version="1">12345</PMID>      
```
i.e. the XML declaration, a pointer to the DTD describing the data, and then the XML data (starting from `<PubmedArticleSet>`) containing the actual data. The DTD is essential for Bio.Entrez to be able to parse the data.

In your case, however, the DTD is missing:
```
>>> handle = Entrez.efetch(db="biosample", id="SAMEA3502652", rettype="xml", retmode="text")
>>> data = handle.read()
>>> print(data[:100].decode())
<?xml version="1.0" ?>
<BioSampleSet><BioSample access="controlled-access" publication_date="2015-04
```
you will see that the XML does not contain a reference to a DTD, but goes into the XML data directly. Then the parser does not know what to do with the XML data, which is why you get an error message if you use
```
>>> Entrez.read(handle, validate=True)
Traceback (most recent call last):
...
Bio.Entrez.Parser.ValidationError: Failed to find tag 'BioSampleSet' in the DTD. To skip all tags that are not represented in the DTD, please call Bio.Entrez.read or Bio.Entrez.parse with validate=False.
```
but the actual problem is that there is no DTD to begin with. Then if instead you use
```
>>> Entrez.read(handle, validate=False)
```
then effectively the parser will skip all XML data, and when it gets to the end of the XML it realizes there is no record to return, hence the error.

We should modify the parser to show a different error message if there is no DTD whatsoever; the `validate=False` option is only useful if there is a DTD, but some elements in the XML are missing from the DTD.

**Comment by camhilker** [comment]
Thank you for your help! 

One more question: Do the biosample IDs have to begin with 'SAMN---'? I was expecting to find a record for a soil metagenome sample for ID: SAMEA3502652, but I noticed that it is changed to SAMN3502652 in the output, and the information is for a different sample. Thanks again!

**Comment by mdehoon** [comment]
When I try, I am also getting SAMN3502652.
You can get the exact URL that biopython is using from the handle:
```
>>> from Bio import Entrez
>>> handle = Entrez.efetch(db="biosample", id="SAMEA3502652", rettype="xml", retmode="text")
>>> handle.url
'https://eutils.ncbi.nlm.nih.gov/entrez/eutils/efetch.fcgi?db=biosample&id=SAMEA3502652&rettype=xml&retmode=text&tool=biopython'
```
If you use this exact URL in a web browser, you will again get SAMN3502652. If you remove `&rettype=xml&retmode=text&tool=biopython` from the URL, you will get SAMN3502652 in plain text format. So this looks like a problem on NCBI's side. 

Apparently more people have run into this. See this link for a workaround:
https://www.biostars.org/p/235280/

**Comment by mdehoon** [comment]
Fixed by #2820
